### PR TITLE
Port parse tests

### LINKED ISSUE #393: Port Parsing tests to the new test format

### PR DESCRIPTION
**Pull request by lloydk**
Closes #393

Shoutout to Github Copilot!

### CODE REVIEW

File: test/parse.js [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+					args: "lch(100 0 none)",
+					expect: '{"spaceId":"lch","coords":[100,0,null],"alpha":1}'
+				}
+			]
+		},
+		{
+			name: "Oklab colors",
+			tests: [
+				{
+					name: "oklab(100% 0 0)",
```
Comment (LeaVerou) [inline test/parse.js:139]:
```suggestion
```

(`name` automatically defaults to the first argument)

File: test/parse.js [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+					args: "color(display-p3 none 1 .5)",
+					expect: '{"spaceId":"p3","coords":[null,1,0.5],"alpha":1}'
+				}
+			]
+		},
+		{
+			name: "hsl()",
+			tests: [
+				{
+					name: "hsl(), commas",
```
Comment (LeaVerou) [inline test/parse.js:364]: Does the output not show the parent name before the child name or in a way that implies containment?

File: test/parse.js [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+					args: "hsl(900, 50%, 50%)",
+					expect: '{"spaceId":"hsl","coords":[900,50,50],"alpha":1}'
+				},
+				{
+					name: "hsla(), degrees for hue, spaces and slash",
+					args: "hsl(90deg 0% 0% / .5)",
+					expect: '{"spaceId":"hsl","coords":[90,0,0],"alpha":0.5}'
+				},
+				{
+					name: "hsla(), rad for hue, spaces and slash",
```
Comment (LeaVerou) [inline test/parse.js:384]: This (and some of the ones after) mention `hsla()` but  use `hsl()`, is that what the original tests did?

**Review by LeaVerou — APPROVED** [review]
Hey, thanks so much! Yes, Copilot was a LIFESAVER when porting these. 
See a few comments, but none of them are blockers.

File: test/parse.js (hunk 2 above)
Comment (lloydk) [reply test/parse.js:359]: It does, the function portion of the name is redundant

File: test/parse.js (hunk 3 above)
Comment (lloydk) [reply test/parse.js:379]: That's what the original tests did.

**Comment by lloydk** [comment]
I'm going to leave the redundant/incorrect hsl functions in the test names for now.

I don't think hsl is spec compliant so I'll open a new issue for that (#428). The tests can be corrected as part of dealing with that issue.

**Pull request merged**